src/Microsoft.Data.SqlClient/tests/FunctionalTests/SqlConnectionBasicTests.cs

```diff
@@ -23,6 +25,35 @@ public void ConnectionTest()
             connection.Open();
         }
 
+        [Fact]
+        public void ConnectionTestWithCultureTH()
```

Comment:
Are you sure this test touches the `InferNamedPipesInformation` method?

Reply:
Yes, in .netcore3.1 it goes into InferNamedPipesInformation and returns true to the calling function even when the data source is not named pipe because of the StartsWith issue with {th-TH} culture.